ui/src/common/components/CitationSummaryGraph/CitationSummaryGraph.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { Component, Fragment } from 'react';
+import {
+  XAxis,
+  YAxis,
+  HorizontalGridLines,
+  VerticalBarSeries,
+  LabelSeries,
+  FlexibleWidthXYPlot,
+  DiscreteColorLegend,
+} from 'react-vis';
+import PropTypes from 'prop-types';
+import { Map } from 'immutable';
+import { Row, Col } from 'antd';
+import './CitationSummaryGraph.scss';
+import styleVariables from '../../../styleVariables';
+
+class CitationSummaryGraph extends Component {
+  LEGEND_ITEMS = [
+    { title: 'Citeable', color: styleVariables['primary-color'] },
+    { title: 'Published', color: styleVariables['orange-6'] },
+  ];
+
+  keyToLabel = new Map({
+    '0.0-1.0': '0',
+    '1.0-10.0': '1-9',
+    '10.0-50.0': '10-49',
+    '50.0-100.0': '50-99',
+    '100.0-250.0': '100-249',
+    '250.0-500.0': '250-499',
+    '500.0-*': '500+',
+  });
+
+  static getSeriesData(data) {
+    const result = [];
+    data.forEach(bucket => {
+      const docCount = bucket.get('doc_count');
+      const countLabel = docCount !== 0 ? docCount.toString() : null;
+      result.push({ x: bucket.get('key'), y: docCount, label: countLabel });
+    });
+    return result;
+  }
+
+  render() {
+    const { citationSummary } = this.props;
+    const citeableData = citationSummary.getIn([
+      'citations',
+      'buckets',
+      'all',
+      'citation_buckets',
+      'buckets',
+    ]);
+    const publishedData = citationSummary.getIn([
+      'citations',
+      'buckets',
+      'published',
+      'citation_buckets',
+      'buckets',
+    ]);
+    const publishedSeriesData = CitationSummaryGraph.getSeriesData(
+      publishedData
+    );
+    const citeableSeriesData = CitationSummaryGraph.getSeriesData(citeableData);
+
+    const yDomainMax = Math.max(
```

Comment:
if those are `immutable.List` not `array`, `maxBy` can be used. But we'll end up converting to JS before that probably.

Reply:
I don't like 1 extra looping over the whole data and recreating new arrays multiple times. We could use `loadash.maxby` and do `Math.max(maxBy(published one), maxBy(citetable one))`

Reply:
you could alternatively get rid of checks by doing:
```
const maxPublished = maxBy(publishedSeriesData, 'y')
...
const yDomainMax = maxBy([maxPublished, maxCiteable], 'y').y
```

Reply:
maxBy has this weird behavior where it returns undefined if it can't find the iteratee (y in this case). So, this wouldn't work.

Reply:
ahh true, you still need a check for the last one :/